### PR TITLE
[New] Manually add timbre w/ npm auto-update

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/mohayonao/timbre.js
**Watch 59
  Star 611
 Fork 50**
@Piicksarn could u help me check this PR for #6110 ? thank you!
- [x] Not found on cdnjs repo
- [x] No already exist issue and PR
- [x] Notable popularity
- [x] Project has public repository on famous online hosting platform (or been hosted on npm) 
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct

### CODE REVIEW

**Comment by Piicksarn** [comment]
@PeterDaveHello, I think the pr is fine!
would you mind to have a look at the pr?
Thank you!

**Comment by maruilian11** [comment]
@Piicksarn i have rebased it, could you help me check it again?

**Comment by maruilian11** [comment]
@amomo could you help me check this PR? thank you!

**Comment by Amomo** [comment]
@maruilian11 
According to its [README.md](https://github.com/mohayonao/timbre.js/blob/master/README.md#nodejs), I think `"timbre.node.js"` is a node module.
We should add "timbre.dev.js", not "timbre.node.js".

File: ajax/libs/timbre/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/mohayonao/timbre.js.git",
+    "basePath": "",
+    "files": [
+      "timbre.node*.js"
+    ]
+  },
+  "requiredFiles": [
```
Comment (Amomo) [inline ajax/libs/timbre/package.json]: You have point the minified file in "filename" field, no need to add "requiredFiles".

**Comment by maruilian11** [comment]
@Amomo  thank you~ i have modified it~

**Comment by Amomo** [comment]
@PeterDaveHello I think this PR is fine, could you take a look? Thank you! :-)

**Comment by PeterDaveHello** [comment]
Use dev file as main file is strange, any other source we can use to import or download the files?

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by maruilian11** [comment]
@PeterDaveHello ping the author in the issue.
https://github.com/mohayonao/timbre.js/issues/57#issuecomment-287617592

**Comment by maruilian11** [comment]
@PeterDaveHello I use `Gruntfile.js` to generate `timbre.js` and `timbre.js.map`, and find `timbre.js` source is `timbre.dev.js`. Since the author has no replied for a long time, can we directly add `timbre.js` and `timbre.dev.js` in this case?

**Comment by PeterDaveHello** [comment]
We can manually build and add it from tagged commit in this situation.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer I manually add versions of timbres, please help me review this PR, thank you!

**Comment by sufuf3** [comment]
The detail commit message maybe can mention that build v13.03.01 ~ v13.05.01 and v14.11.25 of timbre.js, and add timbre.dev.js from git tags on GitHub.
Such as 
```
Manually add timbre v13.03.01 ~ v14.11.25
Manually build v13.03.01 ~ v13.05.01 and v14.11.25 of timbre.js, and add timbre.dev.js from git tags on GitHub.
Add v14.10.12 of timbre.js and timbre.dev.js from offical website(https://mohayonao.github.io/timbre.js)
cc @mohayonao, close #6110
```
Because after I search for a while, I finally know that you didn't build two of files but just build timbre.js.

What do you think?

Thank you.

**Comment by dakshshah96** [comment]
@sufuf3 I don't know. Sounds like a lot for a commit message. I was looking forward to keeping the commit message simple and small.

Let's go with whatever is the consensus among all maintainers finally 😄

**Comment by sufuf3** [comment]
I think yours is fine, also. 
Just need to wrapped to 72 characters according to https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project
And if I can understand from the message that this commit just build timbre.js and add timbre.dev.js from git tags on GitHub. It will be useful to me to know the detail.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer @sufuf3 @dakshshah96 Is that ok?

**Comment by PeterDaveHello** [comment]
Ping @extend1994

**Comment by maruilian11** [comment]
@extend1994 Can you install `timbre` npm package successfully? I can't. About the homepage issue, the homepage domain may change, put the url will help us trace the source in the future.

**Comment by PeterDaveHello** [comment]
@maruilian11 you don't need to "install" it.

**Comment by extend1994** [comment]
@maruilian11 I failed though I tried the author's response to you https://github.com/mohayonao/timbre.js/issues/50. Maybe you can try it again?

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer I add versions on git repo and npm package. And add npm auto-update for future usage. Please help me review this PR again. Thanks.

**Comment by PeterDaveHello** [comment]
```
Manually add them from git repo because timbre.js need to be built
manually and these versions only on git repo. cc #6110
```
```
Manually add them from npm package because timbre.js and
timbre.js.map need to be built manually.
```

1) `need` -> `needs`
2) `and these versions only on git repo`: can't understand at all, is this a completed sentence?

So after all, how did you add it? You built them, or you fetched them from npm/GitHub? Confusing here...

**Comment by maruilian11** [comment]
I downloaded the git repo and npm package. Built the files manually with `Makefile` or `Gruntfile` depended on different versions.

**Comment by maruilian11** [comment]
@PeterDaveHello Is the explanation acceptable?

File: ajax/libs/timbre/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "timbre",
+  "version": "14.6.23",
```
Comment (PeterDaveHello) [inline ajax/libs/timbre/package.json]: Shouldn't this be `14.11.25`?

**Comment by maruilian11** [comment]
@PeterDaveHello Sorry for mistaking the latest version number. I have modified it.

**Comment by sufuf3** [comment]
@maruilian11 
1. I can't find `timbre.js` and `timbre.js.map` of timbre@14.11.25 from npm package, how do you get them? Did you build them? If yes, how about mention this in the commit message? What do you think?
2. How about download `timbre.js` and `timbre.js.map` of v14.10.12 from http://mohayonao.github.io/timbre.js/? let the files as same with the website. What do you think?
Thank you.

**Comment by maruilian11** [comment]
@sufuf3 The website doesn't provide the map file, to maintain the file structure, I think we had better to use the files built from npm v14.10.12.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer I updated the commit msg, please help me review the PR, thanks.

**Comment by PeterDaveHello** [comment]
@sufuf3 why are you starting to duplicate commit message again?

**Comment by PeterDaveHello** [comment]
So the second and the third commit message says the files need to be built from Gruntfile.js manually, what's the relationship with the npm and git you mentioned? So are they built by you? Confusing message here.

**Comment by sufuf3** [comment]
Sorry I just think the second commit message and detail commit message both of them start with `Manually add`, so I think maybe we can let it better. (which I learn the past)

Separating the commit messages between second and third, because can mention detailly.

But as https://github.com/cdnjs/cdnjs/pull/6295#issuecomment-319281671 this comment Maybe I don't have to touch them. 😞

File: ajax/libs/timbre/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "synth"
+  ],
+  "npmName": "timbre",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "timbre.dev.js",
+        "timbre.js",
+        "timbre.js.map"
```
Comment (extend1994) [inline ajax/libs/timbre/package.json]: Maybe single line pattern `timbre.!(node*)*.+(js|map|css)`?

Comment (maruilian11) [reply ajax/libs/timbre/package.json]: The reason I use a complete file name is the author hasn't put `timbre.js` and `timbre.js.map` to the repo. So I think it's better we fetch the file we want certainly.

**Comment by maruilian11** [comment]
@PeterDaveHello I meet some problem when using `Makefile` in `13.02.06`, can I minify `timbre.dev.js` by myself?

**Comment by PeterDaveHello** [comment]
@maruilian11 why don't you mention the problem first?

**Comment by maruilian11** [comment]
```
unama@server ~/timbre.js $ uglifyjs --unsafe -nc -nm --source-map ./timbre.js.map -o ./timbre.js ./timbre.dev.js
Supported options:
  content         null
  filename        null
  includeSources  false
  root            null
  url             null
ERROR: `./timbre.js.map` is not a supported option
    at DefaultsError.Object.defineProperty.get (eval at <anonymous> (/home/unama/.nvm/versions/node/v4.8.3/lib/node_modules/uglify-js/tools/node.js:21:1), <anonymous>:79:23)
    at fatal (/home/unama/.nvm/versions/node/v4.8.3/lib/node_modules/uglify-js/bin/uglifyjs:268:52)
    at run (/home/unama/.nvm/versions/node/v4.8.3/lib/node_modules/uglify-js/bin/uglifyjs:225:9)
    at Object.<anonymous> (/home/unama/.nvm/versions/node/v4.8.3/lib/node_modules/uglify-js/bin/uglifyjs:155:5)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:140:18)
```
I use `make` command first, then directly use command in `Makefile`.

**Comment by PeterDaveHello** [comment]
@maruilian11 I'll take a look later. @sufuf3 please help verify this issue first, thanks.

**Comment by sufuf3** [comment]
Same problem.
I think something wrong with timbre.js.map
If I remove `--source-map ./timbre.js.map`, then it's work.

File: ajax/libs/timbre/package.json (hunk 3 above)
Comment (PeterDaveHello) [reply ajax/libs/timbre/package.json]: I didn't see a need to use multi pattern here ... using the flexible method is preferred.

**Comment by PeterDaveHello** [comment]
That's funny ... isn't this simply working? `uglifyjs --source-map -o timbre.js timbre.dev.js`

I think it clearly told you the problem, didn't it? `ERROR: `./timbre.js.map` is not a supported option`

So you both stock on the problem that you don't take a clear look at the error message ... 

Take a look at the help message and I didn't see any of them telling to specify the filename of map file
```
peter $ uglifyjs --help

  Usage: uglifyjs [options] [files...]


  Options:

    -V, --version                output the version number
    -p, --parse <options>        Specify parser options.
    -c, --compress [options]     Enable compressor/specify compressor options.
    -m, --mangle [options]       Mangle names/specify mangler options.
    --mangle-props [options]     Mangle properties/specify mangler options.
    -b, --beautify [options]     Beautify output/specify output options.
    -o, --output <file>          Output file (default STDOUT).
    --comments [filter]          Preserve copyright comments in the output.
    --config-file <file>         Read minify() options from JSON file.
    -d, --define <expr>[=value]  Global definitions.
    --ie8                        Support non-standard Internet Explorer 8.
    --keep-fnames                Do not mangle/drop function names. Useful for code relying on Function.prototype.name.
    --name-cache <file>          File to hold mangled name mappings.
    --self                       Build UglifyJS as a library (implies --wrap UglifyJS)
    --source-map [options]       Enable source map/specify source map options.
    --timings                    Display operations run time on STDERR.
    --toplevel                   Compress and/or mangle variables in toplevel scope.
    --verbose                    Print diagnostic messages.
    --warn                       Print warning messages.
    --wrap <name>                Embed everything as a function with “exports” corresponding to “name” globally.
    -h, --help                   output usage information

peter $ uglifyjs -V
uglify-js 3.1.1
```

**Comment by maruilian11** [comment]
@PeterDaveHello I think @sufuf3 looked the error msg and find the problem. So you may misunstand her.

**Comment by maruilian11** [comment]
>Anyway, I didn't see a need to use Makefile to make the non-minified dist file in version 13.02.06 as it's already there, isn't it? @maruilian11 ?

@PeterDaveHello  Sorry, I don't get your meaning. I didn't use Makefile to make the non-minified dist file, but to produce a official file of it.

**Comment by PeterDaveHello** [comment]
@maruilian11 The file is already there, isn't it?

**Comment by maruilian11** [comment]
@PeterDaveHello Yes, `timbre.dev.js` is already there, but `timbre.js` and `timbre.js.map` need to be produced.

**Comment by PeterDaveHello** [comment]
So you don't need to deal with Makefile, do you?

To summarize the situation here, this PR is waiting for some versions need to be added manually and those versions need manually minification also? If so, can we simply use git ignore them to make the case simple?

**Comment by maruilian11** [comment]
@PeterDaveHello I don't get it. You mean we minify ourself? Not through the `Makefile`?

**Comment by PeterDaveHello** [comment]
@maruilian11 Does Makefile contain the minification part? I don't think so ...

**Comment by maruilian11** [comment]
@PeterDaveHello Yes, it does. 
Following content is from timbre@13.1.18 Makefile in npm package:
>gh-pages: clear
    @uglifyjs --unsafe -nc -nm -o ./timbre.js ./timbre.dev.js

**Comment by PeterDaveHello** [comment]
Okay, fine, it's part of `gh-pages` which is not a common case.

**Comment by maruilian11** [comment]
@PeterDaveHello The test seems not working. Please take a look.

**Comment by sashberd** [comment]
I have tried to rebase and push tis commit in order to send it again to check. The rebase was succeed, but push provides me next message: 
"To https://github.com/maruilian11/cdnjs.git
 ! [remote rejected]         timbre -> timbre (permission denied)
error: failed to push some refs to 'https://github.com/maruilian11/cdnjs.git'
"

@extend1994 @sufuf3 @PeterDaveHello

**Comment by sashberd** [comment]
@PeterDaveHello Is this PR only need to be rebased and then merged?

**Comment by sashberd** [comment]
Moved to #13700

**Pull request closed without merging**